dev_docs/best_practices.mdx

```diff
@@ -0,0 +1,148 @@
+---
+id: kibBestPractices
+slug: /kibana-dev-docs/best-practices
+title: Best practices
+summary: Best practices to follow when building a Kibana plugin.
+date: 2021-03-17
+tags: ['kibana', 'onboarding', 'dev', 'architecture']
+---
+
+## General
+
+First things first, be sure to review our <DocLink id="kibDevPrinciples" text="development principles"/> and check out all the available
+platform <DocLink id="kibBuildingBlocks" text="building blocks"/> that can simplify plugin development.
+
+## Performance
+
+Build with scalability in mind.
+
+- Consider data with many fields
+- Consider data with high cardinality fields
+- Consider large data sets, that span a long time range
+- Consider slow internet and low bandwidth environments
+
+## Accessibility
+
+Did you know Kibana makes a public statement about our commitment to creating an accessible product for people
+with disabilities? [We do](https://www.elastic.co/guide/en/kibana/master/accessibility.html)! It's very important
+all of our apps are accessible.
+
+- Learn how [EUI tackles accessibility](https://elastic.github.io/eui/#/guidelines/accessibility)
+- If you don't use EUI, follow the same EUI accessibility standards
+
+<DocCallOut title="Internal only">
```

Comment:
I've been maintaining an internal guide on a11y here: https://github.com/elastic/dev/blob/master/ui/accessibility.md

Should we either consolidate everything here or cross-link to it?

(There's not _much_ in there that is internal-only but some is...)

Reply:
If this guide is valid at the stack/dev org level, I think it would make sense to move to a new Mission called `Elastic Developer Guide`. This would be public. I think it's okay if there is information that is relevant only to internal folks, but not if there is information that _needs_ to be kept internal only.  I haven't yet set up this Elastic developer-org wide guide, however.